clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MergeConflictResolver.java

```diff
@@ -617,32 +616,35 @@ private GenericRecord deserializeWriteComputeBytes(
   private ValueAndRmd<GenericRecord> prepareValueAndRmdForUpdate(
       ByteBuffer oldValueBytes,
       RmdWithValueSchemaId rmdWithValueSchemaId,
-      SchemaEntry readerValueSchemaEntry) {
+      SchemaEntry superSetSchemaSchemaEntry) {
 
     if (rmdWithValueSchemaId == null) {
       GenericRecord newValue;
       if (oldValueBytes == null) {
         // Value and RMD both never existed
-        newValue = AvroSchemaUtils.createGenericRecord(readerValueSchemaEntry.getSchema());
+        newValue = AvroSchemaUtils.createGenericRecord(superSetSchemaSchemaEntry.getSchema());
       } else {
         /**
-         * RMD does not exist. This means the value is written in Batch phase and does not have RMD associated. In this
-         * case, the value must be retrieved from storage engine, and is prepended with schema ID.
+         * RMD does not exist. This means the value is written in Batch phase and does not have RMD associated. Records
+         * should have the schema id in the first few bytes unless they are assembled from a chunked value. In order
```

Comment:
So is it the case that `oldValueBytes` would be the fully assembled payload, in case of chunking? So this would never be a manifest nor a chunk?

It is concerning if the schema ID can either be present or not at the start of the `oldValueBytes`... because we need to either treat the payload as going from 0 to len -1, or from 4 to len -1, depending on the case, right?

Reply:
Right.  It is never a manifest or chunk.  It's the real McCoy

Reply:
oldValueBytes are already retrieved via chunking adapter. It is just it does not come with schema id prepended afaik. Having it prepended is another way of fixing it, but if superset schema works then it is even simpler.

Reply:
Actually, I did a test, deserialization with superset schema being writer schema will fail..

Reply:
tweaked it to instead pass along the schemaid from the chunk manifest.